guidelines/commit_and_pr_text.md

```diff
@@ -0,0 +1,56 @@
+
+* Try to describe the problem the user sees, rather than the code change you are doing
+  * I.e. “Console does not work when user clicks X” instead of “Fix drawer notification integration with underlying controller”
+* Prefix bug fixes with the bug number, “Bug XXXXX:” 
+  * (capitalize bug, space, number, then colon)
+* Do not let your PR title get truncated/wrapped into the description
+* Additional prefixes like what release the PR applies to are allowed, but they will be stripped from the changelog (the changelog for 4.4 doesn’t need to say “release-4.4” 
+  * Changelog strips anything between square brackets that has no spaces - i.e. [release-4.4] is stripped but [release 4.4] would not be
+    
+
+__PR description:__  Additional context reviewers need to understand your PR
+
+* Why was this change needed?
+* What changes are being made?  
+* Reference to any prior work(other PRs) or follow up work that will be needed.
+
+[Canonical PR text example](https://github.com/openshift/openshift-apiserver/pull/69)
+
+# Git Commit Best practices
+
+## Number of Commits
+
+Minimize your number of commits per PR with a goal of a single commit in many cases.
+
+Minimal numbers commits per PR make it easier to review the PR, easier to revert the PR if necessary, easier for someone doing “git blame” to follow the set of involved changes, and encourage smaller “chunks of work” (if it’s multiple commits, think about whether it should also be multiple PRs)
+
+Valid reasons to have multiple commits:
+
+* Separating implementation work from generated code or vendor updates
+* Separating api work from implementation
+* Moving files around in a refactor (do the file move in one commit, edit the files in the second commit, so that the edits stand out in the commit diff)
+* Temporarily adding additional commits in response to reviewer feedback so reviewers can easily see how their feedback was addressed.  These commits should be squashed prior to merging.
+
+Invalid reasons to have multiple commits
+* Separating tests+implementation
+* Committing+pushing your work each day/hour (use amend instead, or at least ensure you squash before merging).
+
+## Git Commit text
+
+__First line:__  An english language sentence with capitalization, no period, max 50 characters
+
+__Additional information:__  Why was this change needed?  What is the gist of the change?
+
+* Almost every commit should have more than just a single line descriptor.  If you are making a commit with no additional information, you should think twice about whether you've really provided enough information that another developer can understand the intent of the commit.
+* Wrap lines at 72 characters, barring exceptions for links or other content that can't be wrapped.
+* Avoid referencing github issues/PRs if you are repeatedly amending the commit as it generates a lot of noise in the referenced issue/PR.
```

Comment:
I find references incredibly valuable, but this caveat does seem valuable.  How about this:

 * Add all relevant references to the commit text, including other commits, bugs, PRs, or other links that help explain or understand the change.  Note however that you should minimize amending a commit with github references, as it will generate a lot of noise on reference issues or PRs.  Instead, consider adding commits while iterating on a PR and squashing them once before merging.

This also seems like "github best practices", maybe to be added up in the github section instead of here?

Reply:
revised